lib/msf/core/exe/segment_appender.rb

```diff
@@ -16,6 +16,12 @@ def payload_stub(prefix)
     def generate_pe
       # Copy our Template into a new PE
       pe_orig = Metasm::PE.decode_file(template)
+      if is_warbird?(pe_orig)
+        raise RuntimeError, "The template to inject to appears to have license verification (warbird)"
+      end
+      if pe_orig.export && pe_orig.export.num_exports == 0
+        raise RuntimeError, "The template file doesn't have any exports to inject into!"
+      end
```

Comment:
@gwillcox-r7 You'll be more experience with PE files than me, but I imagined for appending to a file without any sections then this code should still gracefully add a new section? 🤔

Reply:
@adfoster-r7 Not sure why you are referring to sections here? Unless you mean if a PE file doesn't have a `.exports` section or something similar than we should create one, in which case I would agree however this would involve adding a new feature. For now this is intended more as a patch until such features can be added.

Reply:
Marking as resolved since this change was undone with https://github.com/rapid7/metasploit-framework/pull/14683/commits/bffbf4601735595cddddad35b2d848455ad68854 and there is now a separate PR at https://github.com/rapid7/metasm/pull/1 from Spencer to resolve this issue properly at its source within Metasm.